src/feed.sol

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2020 Centrifuge
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+pragma solidity >=0.5.15;
+
+import "ds-note/note.sol";
+import "tinlake-auth/auth.sol";
+import "tinlake-math/interest.sol";
+import "./nftfeed.sol";
+import "./buckets.sol";
+
+contract Feed is BaseNFTFeed, Interest, Buckets {
+    // nftID => maturityDate
+    mapping (bytes32 => uint) public maturityDate;
+
+    // risk => recoveryRatePD
+    mapping (uint => uint) public recoveryRatePD;
+
+    // loan => futureValue
+    mapping (uint => uint) public futureValue;
+
+    WriteOff [2] public writeOffs;
+
+    struct WriteOff {
+        uint rateGroup;
+        // denominated in RAY (10^27)
+        uint percentage;
+    }
+
+    uint public discountRate;
+    uint public maxDays;
+
+    constructor (uint discountRate_, uint maxDays_) public {
+        discountRate = discountRate_;
+        maxDays = maxDays_;
+    }
+
+    function init() public {
+        super.init();
+        // gas optimized initialization of writeOffs
+        // write off are hardcoded in the contract instead of init function params
+
+        // risk group recoveryRatePD
+        recoveryRatePD[0] = ONE;
+        recoveryRatePD[1] = 90 * 10**25;
+
+        // 60% -> 40% write off
+        writeOffs[0] = WriteOff(91, 6 * 10**26);
```

Comment:
would be helpful to have a comment explaining the magic numbers 90 & 91 🙈

Reply:
Yes, we anyway should add additional comments explaining the entire calculation